### PR TITLE
Add ScanR structures search engine

### PR DESCRIPTION
**Pull request by mmuman**
In theory ScanR should also search for projects but the API is different,
so we'd need another engine.

### CODE REVIEW

File: searx/engines/scanr_structures.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        content = None
+        if 'highlights' in result:
+            content = result['highlights'][0]['value']
+
+        # append result
+        results.append({'url': url + 'structure/' + result['id'],
+                        'title': result['label'],
+                        # 'thumbnail': thumbnail,
+                        'img_src': thumbnail,
+                        'content': content})
```
Comment (asciimoo) [inline searx/engines/scanr_structures.py]: please use `searx.utils.html_to_text` or `cgi.escape` to escape the content

Comment (mmuman) [inline searx/engines/scanr_structures.py]: Oh I didn't notice searx did its own highlighting of the search term in a different way.

**Comment by mmuman** [comment]
Sh*, unit tests :D

**Comment by mmuman** [comment]
Should be working now.

**Comment by kvch** [comment]
:+1:

**Comment by asciimoo** [comment]
cool, thanks

**Pull request merged**